src/main/java/com/kobylynskyi/graphql/codegen/mapper/TypeDefinitionToDataModelMapper.java

```diff
@@ -80,22 +80,46 @@
      * @param document       Parent GraphQL document
      * @return Freemarker data model of the GraphQL type
      */
-    private static Set<ParameterDefinition> getFields(MappingConfig mappingConfig,
-                                                      ExtendedObjectTypeDefinition typeDefinition,
-                                                      ExtendedDocument document) {
-        // this includes parameters from base definition and extensions
-        List<ParameterDefinition> typeParameters = FieldDefinitionToParameterMapper.mapFields(mappingConfig,
-                typeDefinition.getFieldDefinitions(), typeDefinition.getName());
-        List<ParameterDefinition> typeParametersFromInterfaces = getInterfacesOfType(typeDefinition, document)
-                .stream()
+    private static Collection<ParameterDefinition> getFields(MappingConfig mappingConfig,
+                                                             ExtendedObjectTypeDefinition typeDefinition,
+                                                             ExtendedDocument document) {
+        // using the map to exclude duplicate fields from the type and interfaces
+        Map<String, ParameterDefinition> allParameters = new LinkedHashMap<>();
+
+        // includes parameters from the base definition and extensions
+        FieldDefinitionToParameterMapper.mapFields(mappingConfig, typeDefinition.getFieldDefinitions(), typeDefinition.getName())
+                .forEach(p -> allParameters.put(p.getName(), p));
+        // includes parameters from the interface
+        getInterfacesOfType(typeDefinition, document).stream()
                 .map(i -> FieldDefinitionToParameterMapper.mapFields(mappingConfig, i.getFieldDefinitions(), i.getName()))
                 .flatMap(Collection::stream)
-                .collect(Collectors.toList());
+                .forEach(paramDef -> {
+                    if (allParameters.containsKey(paramDef.getName())) {
+                        ParameterDefinition existingParamDef = allParameters.get(paramDef.getName());
+                        allParameters.put(paramDef.getName(), merge(existingParamDef, paramDef));
+                    } else {
+                        allParameters.put(paramDef.getName(), paramDef);
+                    }
```

Comment:
This is a minor comment, but the whole `if` block can be replaced by:
```java
allParameters.merge(paramDef.getName(), paramDef, ::merge)
```
Or something along those lines.

Reply:
Thank you so much. I never used this method before.